### PR TITLE
Add tests for multiple tours

### LINKED ISSUE #548: Add tests for multiple tours on a page
This will allow us to be certain you can have two tours, as long as only one is running at a time.

Things to test:

- [x] It is possible to style the tours independently 
- [x] The `data-` attributes are prefixed, so they do not interfere with each other
- [x] The tour active classes, etc are prefixed, so it knows which tour to interact with
- [x] The modal overlays only show one overlay, even though two exist on the page

### PR DESCRIPTION
**Pull request by chuckcarpenter**
Closes #548 

This branch removes the need for body data attributes and tests to ensure tours can be run independently.

### CODE REVIEW

File: test/cypress/integration/element-targeting.spec.js [hunk 1]
```diff
@@ -9,7 +9,7 @@ describe('Attaching tooltips to target elements in the DOM on each step', () =>
     cy.visit('/test/dummy/', {
       onLoad(contentWindow) {
         if (contentWindow.Shepherd) {
-          return Shepherd = contentWindow.Shepherd;
+          return (Shepherd = contentWindow.Shepherd);
```
Comment (rwwagner90) [inline test/cypress/integration/element-targeting.spec.js]: I'm curious, what are the parentheses for?

Comment (chuckcarpenter) [reply test/cypress/integration/element-targeting.spec.js]: I think this was added from the linter

File: test/cypress/integration/element-targeting.spec.js [hunk 2]
```diff
@@ -43,4 +43,46 @@ describe('Attaching tooltips to target elements in the DOM on each step', () =>
...
+
+    afterEach(() => {
+      firstTour.complete();
+      secondTour.complete();
+    });
+    it('applies default classes only on each individual tour', async function() {
+      firstTour.start();
+      secondTour.start();
+
+      cy.get('.shepherd-element').and(result => {
```
Comment (rwwagner90) [inline test/cypress/integration/element-targeting.spec.js]: I think we can simplify these and do like:

```js
cy.get('.shepherd-element').should('have.length', 2);
```

File: src/js/step.js [hunk 3]
```diff
@@ -2,7 +2,12 @@ import { Evented } from './evented.js';
 import autoBind from './utils/auto-bind.js';
 import { isElement, isFunction, isUndefined } from './utils/type-check.js';
 import { bindAdvance } from './utils/bind.js';
-import { setupTooltip, parseAttachTo, normalizePrefix, uuid } from './utils/general.js';
+import {
+  setupTooltip,
```
Comment (rwwagner90) [inline src/js/step.js]: It looks like we still have formatting changes here

File: src/js/step.js [hunk 4]
```diff
@@ -97,7 +102,9 @@ export class Step extends Evented {
   constructor(tour, options = {}) {
     super(tour, options);
     this.tour = tour;
-    this.classPrefix = this.tour.options ? normalizePrefix(this.tour.options.classPrefix) : '';
+    this.classPrefix = this.tour.options
+      ? normalizePrefix(this.tour.options.classPrefix)
```
Comment (rwwagner90) [inline src/js/step.js]: And here

File: src/js/step.js [hunk 5]
```diff
@@ -205,21 +210,20 @@ export class Step extends Evented {
...
    */
   _createTooltipContent() {
-    let classes = this.options.classes || '';
+    const classes = this.options.classes || '';
     const descriptionId = `${this.id}-description`;
     const labelId = `${this.id}-label`;
 
     const ShepherdElementComponent = new ShepherdElement({
       target: document.body,
-      props:
```
Comment (rwwagner90) [inline src/js/step.js]: And here etc

File: src/js/tour.js [hunk 6]
```diff
@@ -64,10 +64,10 @@ export class Tour extends Evented {
     this.modal = new ShepherdModal({
       target: options.modalContainer || document.body,
       props:
-        {
-          classPrefix: this.classPrefix,
-          styles: this.styles
-        }
+      {
```
Comment (chuckcarpenter) [inline src/js/tour.js]: this is eslint, you'll have to deal @rwwagner90

File: test/unit/tour.spec.js [hunk 7]
```diff
@@ -21,14 +21,14 @@ describe('Tour | Top-Level Class', function() {
     instance.complete();
   });
 
-  describe('constructor', function() {
-    it('creates a new tour instance', function() {
+  describe('constructor', function () {
```
Comment (chuckcarpenter) [inline test/unit/tour.spec.js]: @rwwagner90 should also stay

File: src/js/step.js (hunk 5 above)
Comment (chuckcarpenter) [reply src/js/step.js]: the spacing is an eslint thing

File: test/unit/tour.spec.js [hunk 8]
```diff
@@ -120,37 +120,29 @@ describe('Tour | Top-Level Class', function() {
...
-    describe('.start()', function() {
-      it('starts a tour that is the current active', function() {
+    describe('.start()', function () {
+      it('starts a tour that is the current active', function () {
         instance.start();
 
         expect(instance).toBe(Shepherd.activeTour);
       });
-
-      it('adds the current tour\'s "id" property to the body as a data attribute', function() {
```
Comment (chuckcarpenter) [inline test/unit/tour.spec.js]: removing this is the only functional change

**Pull request merged**